lib/queue.js

```diff
@@ -86,6 +87,10 @@ class Queue extends events_1.EventEmitter {
             }
             return pollItems();
         }
+        function handleCriticalError(err) {
+            self.emit('error', err);
+            return pollItems();
```

Comment:
I would put a small delay here before running again (similar to the `runAgain` function). This will give some spacing between retries and avoid calling the function recursively too many times.

```js
function handleCriticalError (err) {
  self.emit('error', err);
  return Bluebird.delay(100).then(pollItems);
}
```

Reply:
I wouldn't do that because we want to go back to processing messages as soon as possible. IMO, the exponential backoff strategy is useful when you're retrying to process a single message with a problem, but in this case we are processing the entire queue.